### DISCUSSION TITLE
Logging in the constructor for Seq apps

### DISCUSSION THREAD

**Discussion by jcmrva** [datalust/seq-tickets#1903] [General]
I'm curious about what's preventing this.

I'm using `Log.{level}` statements elsewhere in the main class and they work as expected, but this does not.
```
public class MyApp : SeqApp, ISubscribeToAsync<LogEventData>
{
    public IdLookupApp()
    {
        Log.Information("app start");
    }
}
```
throws:
> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Seq.Apps.SeqApp.get_Log()

Something to do with how the class is instantiated?

**Comment by nblumhardt** [comment]
Hi @jcmrva - `Log` is a property of the `SeqApp` class that's set from outside the instance, so during the constructor call it hasn't been initialized yet.

Your best bet is to override `OnAttached()` and put start-up logging in there, instead. Let me know if this helps.

Best regards,
Nick

**Reply by jcmrva** [reply]
That'll work, thanks!
